js/map/mapObject.mjs

```diff
@@ -202,8 +194,45 @@ MapIcon.prototype.onClick = function(clickPos){
     let prevState = window.savedata[classname][this.cell.id];
     window.updateChecklistProgress(null, !prevState, null, this.cell);
     
-    if(this.cell.notes && this.cell.notes.includes("Random")){
-        updateRandomGateCount(!prevState);
-    }
     return true;
+}
+
+//Gate icons are special, so they get their own class.
+function GateIcon(cell){
+    MapIcon.call(this, cell);
+    this.isRandom = cell.notes?.includes("Random") ?? false;
+}
+
+GateIcon.prototype = Object.create(MapIcon.prototype);
+
+GateIcon.prototype.draw = function(ctx, mouseLoc, currentSelection){
+    if(this.isRandom){
+        if(getRandomGateCount() >= 40){
+            //draw in grey? idk.
```

Comment:
we could invert the gate color like so:
![Icon_Gate inverted](https://user-images.githubusercontent.com/47337746/149001638-c0a5fc39-7930-4174-8937-3e356d89b798.png)

Or put in another slightly transparent gate icon like so:
![Icon_Gate_Ghost](https://user-images.githubusercontent.com/47337746/149001911-e82194da-04e0-4d98-88aa-6310cd12d1f7.png)

I remember these being mentioned as ideas to solve the problem before.

Reply:
oh and also, would this be related to and resolve #136 ?

Reply:
Yeah. If you want to work on that, that's a good thing to do.